packages/serverpod/lib/src/server/websocket_request_handlers/helpers/method_stream_manager.dart

```diff
@@ -168,7 +168,10 @@ class MethodStreamManager {
     var closeSubscriptionFutures = outboundStreamContexts.map(
       (c) => c.subscription.cancel().timeout(
             _closeTimeout,
-            onTimeout: () => c.controller.onCancel?.call(),
+            onTimeout: () async {
+              await c.controller.onCancel?.call();
+              return null;
+            } as Future<Null> Function()?,
```

Comment:
Discussion: Should we add a comment here explaining the behavior? Or is our test enough? 

This is kind of funky so might be worth a comment.

Reply:
Good point, I'll add a comment 👍